### PR TITLE
Feature/frontend 1890 - Search

### PR DESCRIPTION
**Pull request by bnlb**
Resolves [FRONTEND-1890](https://algorithmia.atlassian.net/browse/FRONTEND-1890) and [FRONTEND-1891](https://algorithmia.atlassian.net/browse/FRONTEND-1891)

Deployed here:
https://test.algorithmia.com/developers?q=machine+learning

### CODE REVIEW

File: src/store/search.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Vue from 'vue'
+import clone from 'clone'
+import { debounce } from '../utilities/debounce'
+import { searchIndex } from '../utilities/lunr'
+
+export const filter = {
+  NONE: { name: 'All results', value: 'NONE' },
+  API_DOCS: { name: 'API Docs', value: 'API_DOCS' },
+  DEV_CENTER: { name: 'Documentation', value: 'DEV_CENTER' },
```
Comment (langlk) [inline src/store/search.js]: Can we name this `Developer Docs`, same as the section in the side nav?

File: src/typings/uiv.d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+declare module 'uiv' {
```
Comment (langlk) [inline src/typings/uiv.d.ts]: Are we switching to using typescript for this repo's vue files?

Comment (bnlb) [reply src/typings/uiv.d.ts]: I don't think we should update any components defined in this repo to use TS (right now), but if we're going to import components from Synapse then I think it'd be best for both `dev-center` and `web` to be as flexible as possible and accept TS components.

File: src/store/search.js (hunk 1 above)
Comment (bnlb) [reply src/store/search.js]: Good call! Updating.

File: src/typings/uiv.d.ts (hunk 2 above)
Comment (langlk) [reply src/typings/uiv.d.ts]: Sounds good!

File: _layouts/synapse-default.html [hunk 3]
```diff
@@ -86,56 +89,32 @@
     </header>
     <main class="syn-page-main" id="scrollspy-content">
       <div class="syn-container syn-mt-32 syn-mb-48">
-        {{ content | emojify }}
+        <div v-if="areSearchResultsShown">
```
Comment (langlk) [inline _layouts/synapse-default.html]: Very small nitpick: could we change this to something like `showSearchResults` so it's not quite as wordy?

Comment (bnlb) [reply _layouts/synapse-default.html]: Yeah...that one didn't feel quite right but I couldn't think of anything better. I'll update.

**Pull request merged**